src/main/java/de/uni_tuebingen/ub/nppm/servlet/AjaxServlet.java

```diff
@@ -35,6 +44,229 @@ protected void autocomplete(HttpServletRequest request, HttpServletResponse resp
         }
     }
 
+    protected void detecZusatzNamenKommentar(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            String EinzelbelegID = request.getParameter("EinzelbelegID");
+            if (EinzelbelegID == null) {
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            } else {
+                ArrayList<String> outputList = new ArrayList<>();
+                Einzelbeleg lastEinzelbeleg = EinzelbelegDB.getById(Integer.parseInt(EinzelbelegID));
+                String belegform = lastEinzelbeleg.getBelegform();
+
+                List<Einzelbeleg> einzelbelege = EinzelbelegDB.getBelegformList(belegform);
+
+                Map<Integer, List<Integer>> namenkommentarIdToEinzelbelegIdsMap = new HashMap<>();
+                Map<Integer, String> namenkommentarIdToPlemmaMap = new HashMap<>();
+
+                for (Einzelbeleg eb : einzelbelege) {
+                    int einzelbelegId = eb.getId();
+
+                    List<EinzelbelegNamenkommentar_MM> einzelbeleg_hatnamenkommentar_list = EinzelbelegNamenkommentarDB.getByEinzelbelegId(einzelbelegId);
+
+                    for (EinzelbelegNamenkommentar_MM einzelbeleg_hatnamenkommentar : einzelbeleg_hatnamenkommentar_list) {
+                        NamenKommentar namenKommentar = einzelbeleg_hatnamenkommentar.getNamenKommentar();
+                        if (namenKommentar != null) {
+                            int namenkommentarID = namenKommentar.getId();
+                            String plemma = namenKommentar.getpLemma();
+
+                            if (!namenkommentarIdToEinzelbelegIdsMap.containsKey(namenkommentarID)) {
+                                namenkommentarIdToEinzelbelegIdsMap.put(namenkommentarID, new ArrayList<>());
+                            }
+                            namenkommentarIdToEinzelbelegIdsMap.get(namenkommentarID).add(einzelbelegId);
+
+                            if (!namenkommentarIdToPlemmaMap.containsKey(namenkommentarID)) {
+                                namenkommentarIdToPlemmaMap.put(namenkommentarID, plemma);
+                            }
+                        }
+                    }
+                }
+
+                if (namenkommentarIdToEinzelbelegIdsMap.size() != 0) {
+                    StringBuilder sb = new StringBuilder();
+                    int save_namenkommentarID = -1;
+
+                    for (Map.Entry<Integer, List<Integer>> entry : namenkommentarIdToEinzelbelegIdsMap.entrySet()) {
+                        int namenkommentarID = entry.getKey();
+                        List<Integer> einzelbelegIDs = entry.getValue();
+                        String plemma = namenkommentarIdToPlemmaMap.get(namenkommentarID);
+
+                        sb.append("Für Belegform ").append(belegform).append(" wurde Zusatznamen-Kommentar  ").append(plemma).append(" gefunden. ");
```

Comment:
Ideally these texts should be taken from datenbank_texte

Reply:
done